src/renderer/components/status-bar/auto-update-component.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { withInjectables } from "@ogre-tools/injectable-react";
+import { observer } from "mobx-react";
+import assert from "assert";
+import React, { useEffect, useState } from "react";
+import { Spinner } from "../spinner";
+import type { ProgressOfUpdateDownload } from "../../../common/application-update/progress-of-update-download/progress-of-update-download.injectable";
+import progressOfUpdateDownloadInjectable from "../../../common/application-update/progress-of-update-download/progress-of-update-download.injectable";
+import type { DiscoveredUpdateVersion } from "../../../common/application-update/discovered-update-version/discovered-update-version.injectable";
+import discoveredUpdateVersionInjectable from "../../../common/application-update/discovered-update-version/discovered-update-version.injectable";
+import type { UpdateIsBeingDownloaded } from "../../../common/application-update/update-is-being-downloaded/update-is-being-downloaded.injectable";
+import updateIsBeingDownloadedInjectable from "../../../common/application-update/update-is-being-downloaded/update-is-being-downloaded.injectable";
+import type { UpdatesAreBeingDiscovered } from "../../../common/application-update/updates-are-being-discovered/updates-are-being-discovered.injectable";
+import updatesAreBeingDiscoveredInjectable from "../../../common/application-update/updates-are-being-discovered/updates-are-being-discovered.injectable";
+
+interface Dependencies {
+  progressOfUpdateDownload: ProgressOfUpdateDownload;
+  discoveredVersionState: DiscoveredUpdateVersion;
+  downloadingUpdateState: UpdateIsBeingDownloaded;
+  checkingForUpdatesState: UpdatesAreBeingDiscovered;
+}
+
+interface EndNoteProps {
+  version?: string;
+  note: (version: string) => JSX.Element;
+}
+
+const EndNote = ({ version, note }: EndNoteProps) => {
+  const [idling, setIdling] = useState(false);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => setIdling(true), 5000);
+
+    return () => clearTimeout(timerId);
+  });
+
+  if (idling) {
+    return idle();
+  }
+
+  return note(version ?? "");
+};
+
+const DivWithTestId = (text: string) => <div data-testid="auto-update-component">{text}</div>;
```

Comment:
We should use different `test-id` for each case. In that way our unit test will be more strict and prevent me from breaking it accidentally.